lib/tty/prompt/list.rb

```diff
@@ -87,6 +87,41 @@ def default_submit_keys
         %i[space return enter].freeze
       end
 
+      # Ensure that if any EOL char is passed as an action key
+      # then all EOL chars are included (for cross-system compat)
+      # Maintain any custom labels.
+      #
+      # @return [Array[Hash]]
+      #
+      # @api private
+      def ensure_eol_compat(keys)
+        eol_symbols = %i[enter return].sort
+        key_symbols = keys.keys.sort
+        key_intersection = key_symbols & eol_symbols
+
+        case key_intersection
+        when [], eol_symbols
+          keys
+        else
+          eol_label = keys[key_intersection[0]]
+          all_eol_keys = eol_symbols.reduce({}) { |hash, key|
+            hash.merge({key => eol_label})
+          }
```

Comment:
Not sure if there's a better way to write `.to_h` that's still compatible with Ruby `2.0.x`...

Reply:
thank you, that much nicer